src/khepri_path.erl

```diff
@@ -87,30 +87,60 @@
 -type component() :: node_id() | ?ROOT_NODE | ?THIS_NODE | ?PARENT_NODE.
 %% Component name in a path to a node.
 
-%% TODO: Rename to node_path()
--type path() :: [component()].
+-type native_path() :: [component()].
+%% Native path to a node.
+%%
+%% A native path is a list of atoms, binaries and special components.
+
+-type native_pattern() :: [pattern_component()].
+%% Path pattern which may match zero, one or more nodes.
+%%
+%% A native pattern is a list of atoms, binaries, special components and
+%% conditions.
+
+-type unix_path() :: string() | binary().
+%% Unix-like path to a node.
```

Comment:
What would you think about moving some of the moduledoc's explanation on unix paths down to this type? On the one hand it's nice to get the full overview by reading the moduledoc but on the other I would expect to find the full explanation of the unix path syntax here with `unix_pattern/0` explaining the added glob syntax

Reply:
Good idea. I expanded the documentation and moved some things in the types' description as you suggested.